locals.tf

```diff
@@ -3,40 +3,45 @@ locals {
   # General
 
   # get the region from "location", or else from the local config
-  region     = var.location != "" ? regex("^[a-z0-9]*-[a-z0-9]*", var.location) : data.google_client_config.current.region
+  region = var.location != "" ? regex("^[a-z0-9]*-[a-z0-9]*", var.location) : data.google_client_config.current.region
 
   # get the zone from "location", or else from the local config. If none is set, default to the first zone in the region
   is_region  = var.location != "" ? var.location == regex("^[a-z0-9]*-[a-z0-9]*", var.location) : false
   first_zone = length(data.google_compute_zones.available.names) > 0 ? data.google_compute_zones.available.names[0] : ""
   # all_zones  = length(data.google_compute_zones.available.names) > 0 ? join(",", [for item in data.google_compute_zones.available.names : format("%s", item)]) : ""
-  zone       = ( var.location != "" ? (local.is_region ? local.first_zone : var.location) : (data.google_client_config.current.zone == "" ? local.first_zone : data.google_client_config.current.zone) )
-  location   = var.location != "" ? var.location : local.zone
+  zone     = (var.location != "" ? (local.is_region ? local.first_zone : var.location) : (data.google_client_config.current.zone == "" ? local.first_zone : data.google_client_config.current.zone))
+  location = var.location != "" ? var.location : local.zone
 
   # CIDRs/Network
   default_public_access_cidrs          = var.default_public_access_cidrs == null ? [] : var.default_public_access_cidrs
   cluster_endpoint_public_access_cidrs = var.cluster_endpoint_public_access_cidrs == null ? local.default_public_access_cidrs : var.cluster_endpoint_public_access_cidrs
   vm_public_access_cidrs               = var.vm_public_access_cidrs == null ? local.default_public_access_cidrs : var.vm_public_access_cidrs
   postgres_public_access_cidrs         = var.postgres_public_access_cidrs == null ? local.default_public_access_cidrs : var.postgres_public_access_cidrs
 
-  ssh_public_key = ( var.create_jump_vm || var.storage_type == "standard"
-                     ? file(var.ssh_public_key)
-                     : null
-                   )
+  ssh_public_key = (var.create_jump_vm || var.storage_type == "standard"
+    ? can(file(var.ssh_public_key)) ? file(var.ssh_public_key) : var.ssh_public_key != null ? length(var.ssh_public_key) > 0 ? var.ssh_public_key : null : null
```

Comment:
Could parenthesis be added to this expression to help indicate the associations for the
condition ? true_val : false_val
relationships.

Reply:
How would the parens help in clarification? Would you need extra formatting?

Reply:
No extra formatting, only add parens to indicate the grouping similar to this two-level conditional which is simpler than yours:
 test = "condition ? value : (elif-condition ? elif-value : else-value)"

Reply:
If the parens are added we'll need to start testing again given the logic change to verify. Looking to avoid this if possible just for the sake of semantics.

Reply:
Talked with @dhoucgitter on this one and for now the code will stay as is. In the future we'll be looking to help format, if possible, in a way to keep `tf fmt` happy along with the complex logic blocks making more sense to everyone. Thanks again.

Reply:
Reviewed the logic with Thomas, he tried adding parens for grouping. terraform fmt ends up formatting things that make it harder to understand the associations when that is done. Maybe there are more layers to this conditional than fmt is able to deal with. I'm OK with leaving this as for now.